### PR TITLE
Add read_delim_files and read_excel_files to support multiple files.

### PR DESCRIPTION
**Pull request by hidekoji**
# Description

Add read_delim_files and read_excel_files to support multiple files.

# Checklist
Make sure you have performed following items before submitting this pull request.
If not, please describe the reason.  

- [ ] Add test cases for this fix/enhancement
- [ ] Pass devtools::check()
- [ ] Pass devtools::test()
- [ ] Test installing from github
- [x] Tested with Exploratory

### CODE REVIEW

File: R/system.R [hunk 1]
```diff
@@ -2445,6 +2470,34 @@ get_excel_sheets <- function(path){
...
+                   locale = locale,
+                   na = na, quoted_na = quoted_na,
+                   comment = comment, trim_ws = trim_ws,
+                   skip = skip, n_max = n_max, guess_max = guess_max,
+                   progress = progress, with_api_key = with_api_key, .id = "exp.file.id") %>% mutate(exp.file.id = basename(exp.file.id))  # extract file name from full path with basename and create file.id column.
+    id_col <- avoid_conflict(colnames(df), "id")
+    # copy internal exp.file.id to the id column.
+    df[[id_col]] <- df[["exp.file.id"]]
+    # drop internal column and move the id column to the very beginning.
+    df %>% dplyr::select(!!!rlang::syms(id_col), dplyr::everything(), -exp.file.id)
```
Comment (hideaki) [inline R/system.R:2497]: !!rlang::sym(id_col) might be more appropriate.

File: R/system.R [hunk 2]
```diff
@@ -2334,6 +2334,31 @@ download_data_file <- function(url, type){
...
+           detectDates = detecDates,
+           skipEmptyRows = skipEmptyRows,
+           skipEmptyCols = skipEmptyCols,
+           check.names = check.names,
+           tzone = tzone, .id = "exp.file.id") %>% mutate(exp.file.id = basename(exp.file.id)) # extract file name from full path with basename.
+    id_col <- avoid_conflict(colnames(df), "id")
+    # copy internal exp.file.id to the id column.
+    df[[id_col]] <- df[["exp.file.id"]]
+    # drop internal column and move the id column to the very beginning.
+    df %>% dplyr::select(!!!rlang::syms(id_col), dplyr::everything(), -exp.file.id)
```
Comment (hideaki) [inline R/system.R:2359]: !!rlang::sym(id_col) might be more appropriate.

File: R/system.R [hunk 3]
```diff
@@ -2445,6 +2470,34 @@ get_excel_sheets <- function(path){
...
+                              escape_backslash = FALSE, escape_double = TRUE,
+                              col_names = TRUE, col_types = NULL,
+                              locale = readr::default_locale(),
+                              na = c("", "NA"), quoted_na = TRUE,
+                              comment = "", trim_ws = FALSE,
+                              skip = 0, n_max = Inf, guess_max = min(1000, n_max),
+                              progress = interactive(), with_api_key = FALSE) {
+    # set name to the files so that it can be used for the "id" column created by purrr:map_dfr.
+    files <- setNames(as.list(files), files)
+    df <- purrr::map_dfr(files, exploratory::read_delim_file,delim = delim, quote = quote,
```
Comment (hideaki) [inline R/system.R:2485]: Add space before delim arg.

File: R/system.R (hunk 1 above)
Comment (hidekoji) [reply R/system.R]: Changed as suggested.

File: R/system.R (hunk 2 above)
Comment (hidekoji) [reply R/system.R]: Changed as suggested.

File: R/system.R (hunk 3 above)
Comment (hidekoji) [reply R/system.R]: Added space.

**Pull request merged**